discovery-infra/test_infra/tools/assets.py

```diff
@@ -1,49 +1,158 @@
 import json
 import logging
-import os
-from typing import List
+import subprocess as sp
+
+import xmltodict
+
+from typing import List, Union, Generator
+from collections import OrderedDict
 
 from munch import Munch
-from test_infra import consts, utils
+from netaddr import IPNetwork, IPRange, IPAddress
+
+from test_infra import consts
 
 
 class Assets:
-    ASSETS_LOCKFILE_DEFAULT_PATH = "/tmp"
 
-    def __init__(self, assets_file: str, lock_file: str = None):
+    def __init__(self, assets_file: str):
         self._assets_file: str = assets_file
-        self._took_assets: List[Munch] = list()
-        self._lock_file: str = lock_file or os.path.join(
-            self.ASSETS_LOCKFILE_DEFAULT_PATH, os.path.basename(assets_file) + ".lock"
-        )
 
     def get(self) -> Munch:
         logging.info("Taking asset from %s", self._assets_file)
-        with utils.file_lock_context(self._lock_file):
-            with open(self._assets_file) as _file:
-                all_assets = json.load(_file)
-            asset = Munch.fromDict(all_assets.pop(0))
-            with open(self._assets_file, "w") as _file:
-                json.dump(all_assets, _file)
-            self._took_assets.append(asset)
-        logging.info("Taken asset: %s", asset)
-        return asset
-
-    def _release(self) -> None:
-        logging.info("Returning %d assets", len(self._took_assets))
-        logging.debug("Assets to return: %s", self._took_assets)
-        with utils.file_lock_context(self._lock_file):
-            with open(self._assets_file) as _file:
-                all_assets = json.load(_file)
-            all_assets.extend([Munch.toDict(asset) for asset in self._took_assets])
-            with open(self._assets_file, "w") as _file:
-                json.dump(all_assets, _file)
-
-    def release_all(self) -> None:
-        logging.info("Returning all %d assets", len(self._took_assets))
-        self._release()
+        with open(self._assets_file) as _file:
+            all_assets = json.load(_file)
+        # move asset to the end of the list
+        asset_data = all_assets.pop(0)
+        all_assets.append(asset_data)
+        with open(self._assets_file, "w") as _file:
+            json.dump(all_assets, _file)
+        logging.info("Taken asset: %s", asset_data)
+        return Munch.fromDict(asset_data)
 
 
 class NetworkAssets(Assets):
+    """ An assets class that stores values based on the current available
+        resources, in order to allow multiple installations while avoiding
+        conflicts. """
+
     def __init__(self):
         super().__init__(assets_file=consts.TF_NETWORK_POOL_PATH)
+        self._allocated_ips_objects = []
+        self._allocated_bridges = []
+
+    def get(self) -> Munch:
+        asset = super().get()
+        self._verify_asset_fields(asset)
+
+        self._allocated_bridges.clear()
```

Comment:
not sure i understand why we need to clean it every time we ran get?

Reply:
because allocated resources may be freed or newly allocated resources will be discovered until the next check.
I don't think we need to count on previous check.

Reply:
but then what's the point to have this list as class object? just return value from function

Reply:
as for me it looks a little bit strange but i can live with it.